web/src/stream_create_subscribers_data.ts

```diff
@@ -1,43 +1,46 @@
 import {page_params} from "./page_params";
 import * as people from "./people";
+import type {User} from "./people";
 
-let user_id_set;
+let user_id_set: Set<number | undefined>;
 
-export function initialize_with_current_user() {
+export function initialize_with_current_user(): void {
     const current_user_id = page_params.user_id;
-    user_id_set = new Set();
+    user_id_set = new Set<number | undefined>();
     user_id_set.add(current_user_id);
 }
 
-export function sorted_user_ids() {
-    const users = people.get_users_from_ids([...user_id_set]);
-    people.sort_but_pin_current_user_on_top(users);
-    return users.map((user) => user.user_id);
+export function sorted_user_ids(): number[] {
+    const user_ids = [...user_id_set].filter((user_id): user_id is number => user_id !== undefined);
+    const users = people.get_users_from_ids(user_ids);
+    const valid_users = users.filter((user): user is User => user !== undefined);
+    people.sort_but_pin_current_user_on_top(valid_users);
+    return valid_users.map((user) => user.user_id);
 }
 
-export function get_all_user_ids() {
+export function get_all_user_ids(): number[] {
     const potential_subscribers = people.get_realm_users();
     const user_ids = potential_subscribers.map((user) => user.user_id);
     // sort for determinism
     user_ids.sort((a, b) => a - b);
     return user_ids;
 }
 
-export function get_principals() {
+export function get_principals(): (number | undefined)[] {
```

Comment:
Also once we don't have undefined value in `user_id_set`, this function can just return `number[]`.